src/storage/mod.rs

```diff
@@ -657,23 +664,31 @@ impl Storage {
                         ctx.get_isolation_level(),
                         !ctx.get_not_fill_cache(),
                     );
-                    snap_store
-                        .scanner(ScanMode::Forward, options.key_only, None, None)
-                        .and_then(|mut scanner| {
-                            let res = scanner.scan(start_key, limit);
-                            let statistics = scanner.get_statistics();
-                            thread_ctx.collect_scan_count(CMD, statistics);
-                            thread_ctx.collect_read_flow(ctx.get_region_id(), statistics);
-                            res
-                        })
-                        .map_err(Error::from)
-                        .map(|results| {
-                            thread_ctx.collect_key_reads(CMD, results.len() as u64);
-                            results
-                                .into_iter()
-                                .map(|x| x.map_err(Error::from))
-                                .collect()
-                        })
+
+                    let scan_mode = if options.reverse_scan {
+                        ScanMode::Backward
+                    } else {
+                        ScanMode::Forward
+                    };
+
+                    let mut scanner = snap_store.scanner(scan_mode, options.key_only, None, None)?;
+                    let res = if options.reverse_scan {
+                        scanner.reverse_scan(start_key, limit)
+                    } else {
+                        scanner.scan(start_key, limit)
+                    };
+
+                    let statistics = scanner.get_statistics();
```

Comment:
I'm still wondering this case. Previously, we have `collect_scan_count` and `collect_read_flow` in `snap_store.scanner(..).and_then(..)`, but now there is no `and_then`. There should be some differences in edge cases. The rest LGTM.

Reply:
what is the difference?

Reply:
When `snap_store.scanner(..)` returns `Err`, there will be differences. `and_then` only executes when it is `Ok()`.

Reply:
The previous code did nothing when `snap_store.scanner()` returns `Err`, and now it returns directly when `snap_store.scanner()` returns `Err`, they are exactly the same.